packages/fx-core/src/core/middleware/projectMigrator.ts

```diff
@@ -436,16 +437,15 @@ async function migrateMultiEnv(projectPath: string): Promise<void> {
   );
   //projectSettings.json
   const projectSettings = path.join(fxConfig, ProjectSettingsFileName);
+  const configDevJsonFilePath = path.join(fxConfig, "config.dev.json");
+  const envDefaultFilePath = path.join(fx, "env.default.json");
   await fs.copy(path.join(fx, "settings.json"), projectSettings);
-  await ensureProjectSettings(projectSettings, path.join(fx, "env.default.json"));
+  await ensureProjectSettings(projectSettings, envDefaultFilePath);
 
   const appName = await getAppName(projectSettings);
   if (!migrateFromV1) {
     //config.dev.json
-    await fs.writeFile(
-      path.join(fxConfig, "config.dev.json"),
-      JSON.stringify(getConfigDevJson(appName), null, 4)
-    );
+    await fs.writeFile(configDevJsonFilePath, JSON.stringify(getConfigDevJson(appName), null, 4));
```

Comment:
We can move  the logic of "migrate skipAddingSqlUser" to here. 
Cause they are all about the code of generating config.dev.json file.

Reply:
thanks, updated